lang/c++/impl/DataFile.cc

```diff
@@ -146,16 +139,41 @@ void DataFileWriterBase::sync() {
     } else if (codec_ == DEFLATE_CODEC) {
         std::vector<char> buf;
```

Comment:
We can make `buf` a `std::vector<uint8_t>` and avoid `reintertret_cast` at the end.

Reply:
Fixed. Thanks for your review!